backends/ebpf/psa/externs/ebpfPsaDigest.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2022-present Orange
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "ebpfPsaDigest.h"
+
+#include "backends/ebpf/ebpfType.h"
+#include "backends/ebpf/psa/ebpfPsaDeparser.h"
+
+namespace EBPF {
+
+class EBPFDigestPSAValueVisitor : public CodeGenInspector {
+ protected:
+    cstring digestMapInstance;
+    CodeGenInspector *codegen;
+    EBPFType *valueType;
+
+ public:
+    EBPFDigestPSAValueVisitor(P4::ReferenceMap *refMap, P4::TypeMap *typeMap,
+                              cstring digestMapInstance, CodeGenInspector *codegen,
+                              EBPFType *valueType)
+        : CodeGenInspector(refMap, typeMap),
+          digestMapInstance(digestMapInstance),
+          codegen(codegen),
+          valueType(valueType) {}
+
+    void pushExistingElem(const IR::Expression *expr) {
+        builder->emitIndent();
+        builder->appendFormat("bpf_map_push_elem(&%s, &", digestMapInstance);
+        codegen->visit(expr);
+        builder->append(", BPF_EXIST)");
+        builder->endOfStatement(true);
+    }
+
+    // handle expression like: "digest.pack(msg)"
+    bool preorder(const IR::PathExpression *pe) override {
+        pushExistingElem(pe);
+        return false;
+    }
+
+    // handle expression like: "digest.pack(metadata.msg)"
+    bool preorder(const IR::Member *member) override {
+        pushExistingElem(member);
+        return false;
+    }
+
+    // handle expression like: "digest.pack(value)"
```

Comment:
Is this the 'value' part in digest.pack?

Reply:
I've clarified this comment and other about these `preorder`s